### ISSUE TITLE
Support for SAS tokens

### ISSUE THREAD

**Issue by quigg-caroline** [microsoft/CDM#187]
Hey all,
I'm looking into using this SDK for a project. 

One of our requirements is writing to a data lake using a provided SAS URI. Looks like the ADLSAdapter can use a shared key, TokenProvider, or  a client id/secret.

Is there any way to add support for using SAS tokens for the ADLSAdapter or any work arounds for this?

**Comment by miroslavplese** [comment]
Hello, the SAS support was evaluated earlier this year and at the time we found that directory-level SAS authorization was not yet supported by Storage APIs. Specifically, CDM SDK’s ADLS adapter uses directory level operation to obtain list of all files found under a certain path, to perform data partition pattern matching operations, so such requests need to be accepted when SAS auth is used. Because of the discovered limitation, we decided to hold off from implementing SAS support until the Storage APIs gain the needed capability. There was a plan to GA directory-level support in Storage by end of this year, but we haven’t seen announcements yet. We will revisit this feature in January and if things have improved start planning the work to add this feature.
